cmd/cli/serve/serve.go

```diff
@@ -236,6 +236,10 @@ func serve(cmd *cobra.Command) error {
 		// with the other data in the requesterConfig.
 		nodeConfig.RequesterAutoCert = config.ServerAutoCertDomain()
 		nodeConfig.RequesterAutoCertCache = config.GetAutoCertCachePath()
+
+		cert, key := config.GetRequesterCertificateSettings()
+		nodeConfig.RequesterTLSCertificateFile = cert
+		nodeConfig.RequesterTLSKeyFile = key
```

Comment:
Is this actually specific to just the requester node? The compute node also has APIs, right?

If yes, is there any reason to actually limit it to the requester node? Can't we just use it for the whole node, regardless of type?

Reply:
We absolutely should where we use HTTP, but comms between the requester/compute nodes is currently happening over libp2p which secures itself.  And comms between the executors and the compute nodes are across domain sockets. 

tldr; Yes, but in stages and only when we know what we're TLSing

Reply:
I'm talking about the compute node HTTP API endpoints, yes. E.g. https://github.com/bacalhau-project/bacalhau/blob/715951da29483a823d8c7fe7c88da1c831e9812a/pkg/publicapi/server.go#L58-L63

These work for compute nodes as well, right?

Reply:
> These work for compute nodes as well, right?

Currently, yes. And they should be served over TLS, but I don't know just yet whether:

* they should be using requester-node generated certs that as part of using mTLS, or 
* using self-signed certs with really broad/vague caveats/identities and no validity checks, or 
* publicly generated certs (via ACME or CSR) because they have public names

So yes, will add them, but later when we know how they're going to communicate.